### PR TITLE
Statistical t-test for QMC

### PR DESCRIPTION
**Pull request by Atomyka**
T-test checking for consistency of ED & QMC results. 

Tests now run for observables based on magnetisation (|M|, M²) and energy. Still to-do: heat-capacity. 

Still open for discussion: whether or not to run multiple QMCs with different seeds.

### CODE REVIEW

**Comment by Atomyka** [comment]
The relevant commit is the "t-test with binning for single QMC run". Don't know why the other commits landed in this PR. Is it possible to get rid of them @Roger-luo ? :)

**Comment by weinbe58** [comment]
@Atomyka 

> The relevant commit is the "t-test with binning for single QMC run". Don't know why the other commits landed in this PR. Is it possible to get rid of them @Roger-luo ? :)

I actually just figured this out! This is caused because you're merging instead of rebasing. If there are PR that get merged into the master branch after you have forked your branch you should try to rebase instead of merge and GitHub will ignore the commits that are done through the rebasing.

**Comment by weinbe58** [comment]
@Atomyka It looks like when you deleted the `error.jl` and `TFIM.jl` files you did not remove the include statements and so BloqadeQMC will not compile because those files are missing.

Other than that the test looks good! :)

**Comment by Atomyka** [comment]
> @Atomyka It looks like when you deleted the `error.jl` and `TFIM.jl` files you did not remove the include statements and so BloqadeQMC will not compile because those files are missing.
> 
> Other than that the test looks good! :)

Tests failing was actually due to a faulty = sign ;)

But thanks for mentioning the includes - turns out error.jl from the previous PR wasn't being run at all! Fixed now!

File: lib/BloqadeQMC/test/ltfim.jl [hunk 1]
```diff
@@ -94,41 +96,103 @@ expected_values = Dict{Tuple{Bool, Int, Float64}, Dict{String, Float64}}(
...
         end
     end
+
+    # Binning Analysis for observables based on mags
+    B = LogBinner(abs.(mags))
+    println("Correlation time for abs_mag_binned samples")
+    τ_abs = tau(B)
+    println(τ_abs)
+    println("N_eff")
+    N_eff = MCS / τ_abs
```
Comment (emerali) [inline lib/BloqadeQMC/test/ltfim.jl:130]: LogBinner uses a slightly different convention for tau, so N_eff should be `MCS / (2*tau + 1)`

**Comment by emerali** [comment]
We can discuss this a bit more tomorrow, but I realised that it doesn't make sense to do a t-test for jackknife'd estimators, since the jackknife distribution most likely won't be normal. We're gonna have to think a bit more about the most effective way to test those estimators (do we construct error bars using the jackknife histogram? how would we control autocorrelation in this case? etc.), or we may decide to just skip testing those altogether.

**Comment by Roger-luo** [comment]
> The relevant commit is the "t-test with binning for single QMC run". Don't know why the other commits landed in this PR. Is it possible to get rid of them @Roger-luo ? :)

just do a squash before merging this PR. And always remember to squash before merging.

**Comment by Atomyka** [comment]
Closing pull request since LTFIM test is now working but not relevant in the long-term (switching to Rydberg test)

**Pull request closed without merging**